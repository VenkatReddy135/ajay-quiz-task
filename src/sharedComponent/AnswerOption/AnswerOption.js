import React from "react";
import PropTypes from "prop-types";
import Button from "../Button/Button";

const AnswerOption = (props) => {
  return (
    <li className="answerOption" key={props.index} style={{ margin: "2px" }}>
      <Button
        buttonIndex={props.index}
        handleClick={props.onAnswerSelected}
        buttonClass={
          props.selectedAnswer === props.index
            ? "btn btn-success"
            : "btn btn-light"
        }
        buttonValue={props.answerContent}
      />
    </li>
  );
};

AnswerOption.propTypes = {
  index: PropTypes.any,
  selectedAnswer: PropTypes.any,
  onAnswerSelected: PropTypes.any,
};

export default AnswerOption;
