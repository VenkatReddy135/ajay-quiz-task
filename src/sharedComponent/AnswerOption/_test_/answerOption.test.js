/* eslint-disable testing-library/no-node-access */
import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import AnswerOption from "../AnswerOption";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

describe("Answer Options component", () => {
  const renderedData = renderer.create(<AnswerOption />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AnswerOption />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
