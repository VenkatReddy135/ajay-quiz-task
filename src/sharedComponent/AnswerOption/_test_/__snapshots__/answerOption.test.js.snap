// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Options component renders a snapshot 1`] = `
<li
  className="answerOption"
  style={
    Object {
      "margin": "2px",
    }
  }
>
  <button
    className="btn btn-success"
    type="button"
    value=""
  >
    
  </button>
</li>
`;
